backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/75-de.metas.fresh.reports/5714331_sys_onlyActive_M_Cost_CostPrice_Function.sql

```diff
@@ -0,0 +1,214 @@
+      WHERE p.M_Product_Category_ID =
+            getSysConfigAsNumeric('PackingMaterialProductCategoryID'
+                , wh.AD_Client_ID
+                , wh.AD_Org_ID)
+        AND t.MovementDate::date <= p_keydate
+        AND org.ad_org_id = p_AD_Org_ID
+      GROUP BY pa.P_Asset_acct,
+               wh.C_Activity_ID,
+               wh.M_Warehouse_ID,
+               p.M_Product_ID,
+               p.C_UOM_ID,
+               schema.c_acctschema_id,
+               currency.costingprecision,
+               COALESCE(getCurrentCost(p.M_Product_ID,
+                                       uom.c_uom_ID,
+                                       p_keydate::date,
+                                       schema.c_acctschema_id,
+                                       costElement.M_CostElement_ID,
+                                       p_AD_Client_ID,
+                                       p_AD_org_ID), 0::numeric)) dat
+         LEFT OUTER JOIN M_Product p
+                         ON dat.M_Product_ID = p.M_Product_ID AND p.AD_Org_ID IN (p_ad_org_id, 0) -- products on ANY org are also acceptable
+                             AND CASE WHEN p_onlyActiveProducts = 'Y' THEN p.isactive = 'Y' ELSE 1 = 1 END
+         LEFT OUTER JOIN M_Product_Trl pt
+                         ON p.M_Product_ID = pt.M_Product_ID AND pt.ad_language = p_ad_language AND pt.isActive = 'Y' AND
+                            pt.AD_Org_ID = p_ad_org_id
+         LEFT OUTER JOIN M_Warehouse wh
+                         ON dat.M_Warehouse_ID = wh.M_Warehouse_ID AND wh.isActive = 'Y' AND wh.AD_Org_ID = p_ad_org_id
+         LEFT OUTER JOIN C_Activity a ON dat.C_Activity_ID = a.C_Activity_ID AND a.isActive = 'Y'
+         LEFT OUTER JOIN C_ValidCombination vc
+                         ON dat.C_ValidCombination_ID = vc.C_ValidCombination_ID AND vc.isActive = 'Y'
+         LEFT OUTER JOIN C_UOM uom ON dat.C_UOM_ID = uom.C_UOM_ID AND uom.isActive = 'Y'
+         LEFT OUTER JOIN C_UOM_Trl uomt
+                         ON uom.C_UOM_ID = uomt.C_UOM_ID AND uomt.ad_language = p_ad_language AND uomt.isActive = 'Y'
+WHERE p.m_product_id IS NOT NULL
+  AND CASE WHEN p_onlyActiveProducts = 'Y' THEN qty != 0 ELSE 1 = 1 END
+  AND CASE WHEN p_M_Product_ID IS NOT NULL AND p_M_Product_ID > 0 THEN p.M_Product_ID = p_M_Product_ID ELSE 1 = 1 END
+  AND CASE WHEN p_M_Warehouse_ID IS NOT NULL AND p_M_Warehouse_ID > 0 THEN wh.M_Warehouse_ID = p_M_Warehouse_ID ELSE 1 = 1 END
+ORDER BY vc.combination,
+         vc.description,
+         a.name,
+         Wh.name,
+         p.value
+    ;
+
+
+$$
+    LANGUAGE sql STABLE
+;
+
+
+--
+-- select *
+-- from report.M_Cost_CostPrice_Function('2020-09-16',
+--                                       null,
+--                                       null,
+--                                       'Y',
+--                                       'de_DE',
+--                                       1000000,
+--                                       1000000)
```

Comment:
Please, correct the example for how to run the function. As of now, it doesn't fit the number of parameters.

Reply:
please, also correct it in the ddl

Reply:
done.